src/commands.rs

```diff
@@ -3160,12 +3182,48 @@ fn cmd_obslog(ui: &mut Ui, command: &CommandHelper, args: &ObslogArgs) -> Result
     } else {
         for commit in commits {
             template.format(&commit, formatter)?;
+            if let Some(diff_format) = diff_format {
+                show_predecessor_patch(formatter, &workspace_command, &commit, diff_format)?;
+            }
         }
     }
 
     Ok(())
 }
 
+fn show_predecessor_patch(
+    formatter: &mut dyn Formatter,
+    workspace_command: &WorkspaceCommandHelper,
+    commit: &Commit,
+    diff_format: DiffFormat,
+) -> Result<(), CommandError> {
+    let predecessors = commit.predecessors();
+    if !predecessors.is_empty() {
```

Comment:
Style nit: could be `if let` if you like:
`if let Some(predecessor) = predecessors.first()`

Reply:
Much better, thanks.